Code/Framework/AzCore/Tests/Serialization/Json/AnySerializerTests.cpp

```diff
@@ -0,0 +1,281 @@
+            AZStd::any anyT;
+            anyT = instanceT;
+            return AZStd::make_shared<AZStd::any>(anyT);
+        }
+
+        std::optional<AZStd::string_view> GetJSON([[maybe_unused]] const JSONRequestSpec& requestSpec) override
+        {
+            if (requestSpec.jsonStorage == DefaultJSONStorage::Store)
+            {
+                if (requestSpec.objectSupplied == DefaultObjectSuppliedInSerialization::True)
+                {
+                    if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Full)
+                    {
+
+                    }
+                    else if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Partial)
+                    {
+
+                    }
+                    else
+                    {
+                        // fully configured
+                    }
+                }
+                else
+                {
+                    if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Full)
+                    {
+
+                    }
+                    else if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Partial)
+                    {
+
+                    }
+                    else
+                    {
+                        // fully configured
+                    }
+                }
+            }
+            else
+            {
+                // dropping defaults
+                if (requestSpec.objectSupplied == DefaultObjectSuppliedInSerialization::True)
+                {
+                    if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Full)
+                    {
+
+                    }
+                    else if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Partial)
+                    {
+                        
+                    }
+                    else
+                    {
+                        // fully configured
+                        return m_jsonForFullyConfiguredInstanceWithoutDefaults.begin();
+                    }
+                }
+                else
+                {
+                    if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Full)
+                    {
+
+                    }
+                    else if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Partial)
+                    {
+
+                    }
+                    else
+                    {
+                        return m_jsonForFullyConfiguredInstanceWithoutDefaults.begin();
+                    }
+                }
+            }
+
+            return AZStd::nullopt;
+        }
+
+        AZStd::string_view GetJsonForPartialDefaultInstance() override
+        {
+            return m_jsonForPartialDefaultInstanceStrippedDefaults.begin();
+        }
+
+        AZStd::string_view GetJsonForFullySetInstance() override
+        {
+            return m_jsonForFullyConfiguredInstance.begin();
```

Comment:
AZStd::fixed_string has conversion operator for [string_view](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/string/fixed_string.h#L103).

There is a reason for that, because it quickly construct a string_view using the it's member data pointer and the size it contains.

The reason you had to use `.begin()` in the first place, is due to circumventing the fixed_string API by using azsnprintf on it.